contracts/EthereumLightClient.sol

```diff
@@ -87,12 +92,36 @@ contract EthereumLightClient is Ethash, Initializable, PausableUpgradeable {
         _setFirstBlock(storedBlock);
     }
 
+    function _deleteBlock(uint256 blockNo) private {
+        uint256[] storage blockHashes = blocksByHeight[blockNo];
+        for(uint256 j = 0; j < blockHashes.length; j++) {
+            delete blocks[blockHashes[j]];
+        }
+        delete blocksByHeight[blockNo];
+        delete blocksByHeightExisting[blockNo];
+        delete finalizedBlocks[blockNo];
+        delete verifiedBlocks[blockNo];
```

Comment:
@peekpi do we have to delete from the `blockExisting` map as well?

Reply:
yeah, fixed.